kclvm/sema/src/core/symbol.rs

```diff
@@ -0,0 +1,389 @@
+use generational_arena::Arena;
+use indexmap::IndexMap;
+
+use kclvm_error::Position;
+
+use super::package::{PackageDB, PackageInfo};
+
+#[derive(Default, Debug)]
+pub struct KCLSymbolData {
+    pub(crate) values: Arena<ValueSymbol>,
+    pub(crate) packages: Arena<PackageSymbol>,
+    pub(crate) functions: Arena<FunctionSymbol>,
+    pub(crate) attributes: Arena<AttributeSymbol>,
+    pub(crate) schemas: Arena<SchemaSymbol>,
+    pub(crate) type_aliases: Arena<TypeAliasSymbol>,
+    pub(crate) unresolved: Arena<UnresolvedSymbol>,
+    pub(crate) rules: Arena<RuleSymbol>,
+
+    pub(crate) fully_qualified_name_map: IndexMap<String, SymbolRef>,
+}
+
+impl KCLSymbolData {
+    pub fn get_symbol_by_fully_qualified_name(&self, fqn: &str) -> Option<SymbolRef> {
+        self.fully_qualified_name_map.get(fqn).cloned()
+    }
+
+    pub fn get_fully_qualified_name(&self, symbol_ref: SymbolRef) -> String {
+        match symbol_ref.get_kind() {
+            SymbolKind::Schema => {
+                let schema_symbol = self.schemas.get(symbol_ref.get_id()).unwrap();
```

Comment:
Must the `symbol_ref` be among them e.g., `schemas`, `attributes`, ...? Or is there a better way to handle the `unwrap()` function?